Exception in thread "main" org.postgresql.util.PSQLException: ERROR: relation "spacon.migrations" does not exist

Hello Team,

When I try to migrate using the lein script, I have the following error message:

~/docker/test/spatialconnect-server-master/server$ lein migrate
WARNING: seqable? already refers to: #'clojure.core/seqable? in namespace: clojure.core.incubator, being replaced by: #'clojure.core.incubator/seqable?
WARNING: boolean? already refers to: #'clojure.core/boolean? in namespace: clojure.tools.analyzer.utils, being replaced by: #'clojure.tools.analyzer.utils/boolean?
WARNING: boolean? already refers to: #'clojure.core/boolean? in namespace: clojure.tools.analyzer, being replaced by: #'clojure.tools.analyzer.utils/boolean?
WARNING: bounded-count already refers to: #'clojure.core/bounded-count in namespace: clojure.core.async, being replaced by: #'clojure.core.async/bounded-count
2017-03-24 17:44:55.386 DEBUG   [main] --- spacon.db.conn                           : Making db connection to localhost:5432/spacon
2017-03-24 17:44:55.407  INFO   [MLog-Init-Reporter] --- com.mchange.v2.log.MLog                  : MLog clients using slf4j logging.
2017-03-24 17:44:55.463  INFO   [main] --- com.mchange.v2.c3p0.C3P0Registry         : Initializing c3p0-0.9.5-pre9 [built 08-October-2014 03:06:08 -0700; debug? true; trace: 10]
2017-03-24 17:44:59.417 DEBUG   [main] --- spacon.db.conn                           : Running database migration
2017-03-24 17:44:59.418 DEBUG   [main] --- spacon.db.conn                           : Loading database migration config
2017-03-24 17:44:59.464  INFO   [main] --- c.m.v.c.i.AbstractPoolBackedDataSource   : Initializing c3p0 pool... com.mchange.v2.c3p0.ComboPooledDataSource [ acquireIncrement -> 3, acquireRetryAttempts -> 30, acquireRetryDelay -> 1000, autoCommitOnClose -> false, automaticTestTable -> null, breakAfterAcquireFailure -> false, checkoutTimeout -> 0, connectionCustomizerClassName -> null, connectionTesterClassName -> com.mchange.v2.c3p0.impl.DefaultConnectionTester, contextClassLoaderSource -> caller, dataSourceName -> z8kfsx9nc1w661mipoew|3891f024, debugUnreturnedConnectionStackTraces -> false, description -> null, driverClass -> org.postgresql.Driver, extensions -> {}, factoryClassLocation -> null, forceIgnoreUnresolvedTransactions -> false, forceUseNamedDriverClass -> false, identityToken -> z8kfsx9nc1w661mipoew|3891f024, idleConnectionTestPeriod -> 800, initialPoolSize -> 2, jdbcUrl -> jdbc:postgresql://localhost:5432/spacon, maxAdministrativeTaskTime -> 0, maxConnectionAge -> 3600, maxIdleTime -> 1800, maxIdleTimeExcessConnections -> 120, maxPoolSize -> 10, maxStatements -> 0, maxStatementsPerConnection -> 0, minPoolSize -> 2, numHelperThreads -> 3, preferredTestQuery -> null, privilegeSpawnedThreads -> false, properties -> {user=******, password=******}, propertyCycle -> 0, statementCacheNumDeferredCloseThreads -> 0, testConnectionOnCheckin -> false, testConnectionOnCheckout -> false, unreturnedConnectionTimeout -> 0, userOverrides -> {}, usesTraditionalReflectiveProxies -> false ]
**Exception in thread "main" org.postgresql.util.PSQLException: ERROR: relation "spacon.migrations" does not exist**
  Position : 16, compiling:(/tmp/form-init691726267945491079.clj:1:72)
	at clojure.lang.Compiler.load(Compiler.java:7442)
	at clojure.lang.Compiler.loadFile(Compiler.java:7368)
	at clojure.main$load_script.invokeStatic(main.clj:277)
	at clojure.main$init_opt.invokeStatic(main.clj:279)
	at clojure.main$init_opt.invoke(main.clj:279)
	at clojure.main$initialize.invokeStatic(main.clj:310)
	at clojure.main$null_opt.invokeStatic(main.clj:344)
	at clojure.main$null_opt.invoke(main.clj:341)
	at clojure.main$main.invokeStatic(main.clj:423)
	at clojure.main$main.doInvoke(main.clj:386)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.lang.Var.applyTo(Var.java:700)
	at clojure.main.main(main.java:37)
**Caused by: org.postgresql.util.PSQLException: ERROR: relation "spacon.migrations" does not exist
  Position : 16**
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2270)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:1998)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:255)
	at org.postgresql.jdbc2.AbstractJdbc2Statement.execute(AbstractJdbc2Statement.java:570)
	at org.postgresql.jdbc2.AbstractJdbc2Statement.executeWithFlags(AbstractJdbc2Statement.java:420)
	at org.postgresql.jdbc2.AbstractJdbc2Statement.executeQuery(AbstractJdbc2Statement.java:305)
	at com.mchange.v2.c3p0.impl.NewProxyPreparedStatement.executeQuery(NewProxyPreparedStatement.java:430)
	at clojure.java.jdbc$db_query_with_resultset$run_query_with_params__17888.invoke(jdbc.clj:824)
	at clojure.java.jdbc$db_query_with_resultset.invokeStatic(jdbc.clj:834)
	at clojure.java.jdbc$db_query_with_resultset.invoke(jdbc.clj:790)
	at clojure.java.jdbc$query.invokeStatic(jdbc.clj:866)
	at clojure.java.jdbc$query.doInvoke(jdbc.clj:838)
	at clojure.lang.RestFn.invoke(RestFn.java:464)
	at ragtime.jdbc.SqlDatabase.applied_migration_ids(jdbc.clj:43)
	at ragtime.core$migrate_all.invokeStatic(core.clj:43)
	at ragtime.core$migrate_all.invoke(core.clj:32)
	at ragtime.repl$migrate.invokeStatic(repl.clj:49)
	at ragtime.repl$migrate.invoke(repl.clj:34)
	at spacon.db.conn$migrate.invokeStatic(conn.clj:54)
	at spacon.db.conn$migrate.invoke(conn.clj:52)
	at clojure.lang.Var.invoke(Var.java:375)
	at user$eval27117.invokeStatic(form-init691726267945491079.clj:1)
	at user$eval27117.invoke(form-init691726267945491079.clj:1)
	at clojure.lang.Compiler.eval(Compiler.java:6978)
	at clojure.lang.Compiler.eval(Compiler.java:6968)
	at clojure.lang.Compiler.load(Compiler.java:7430)

Seems there is an issue there!! I'm currently using lein version 2.7.1

> lein -v
Leiningen 2.7.1 on Java 1.8.0_121 Java HotSpot(TM) 64-Bit Server VM

Can you please have a look? Thanks in advance,

Mat

Hello Team,

Thanks again for your support (the schema "spacon" was missing). I tried to update spacialconnect-server-dashboard to 0.10.0 version and com.stuartsierra/component to 0.3.2. But my spacialconnect connect server is still not booting. I have now the following error message:


2017-03-27 08:56:51.031  INFO   [main] --- o.a.k.clients.producer.ProducerConfig    : ProducerConfig values: 
	acks = all
	batch.size = 16384
	block.on.buffer.full = false
	bootstrap.servers = null
	buffer.memory = 33554432
	client.id = sc-producer-ceb7ad798f02
	compression.type = none
	connections.max.idle.ms = 540000
	interceptor.classes = null
	key.serializer = class org.apache.kafka.common.serialization.StringSerializer
	linger.ms = 0
	max.block.ms = 60000
	max.in.flight.requests.per.connection = 5
	max.request.size = 1048576
	metadata.fetch.timeout.ms = 60000
	metadata.max.age.ms = 300000
	metric.reporters = []
	metrics.num.samples = 2
	metrics.sample.window.ms = 30000
	partitioner.class = class org.apache.kafka.clients.producer.internals.DefaultPartitioner
	receive.buffer.bytes = 32768
	reconnect.backoff.ms = 50
	request.timeout.ms = 2000
	retries = 0
	retry.backoff.ms = 100
	sasl.kerberos.kinit.cmd = /usr/bin/kinit
	sasl.kerberos.min.time.before.relogin = 60000
	sasl.kerberos.service.name = null
	sasl.kerberos.ticket.renew.jitter = 0.05
	sasl.kerberos.ticket.renew.window.factor = 0.8
	sasl.mechanism = GSSAPI
	security.protocol = PLAINTEXT
	send.buffer.bytes = 131072
	ssl.cipher.suites = null
	ssl.enabled.protocols = [TLSv1.2, TLSv1.1, TLSv1]
	ssl.endpoint.identification.algorithm = null
	ssl.key.password = null
	ssl.keymanager.algorithm = SunX509
	ssl.keystore.location = null
	ssl.keystore.password = null
	ssl.keystore.type = JKS
	ssl.protocol = TLS
	ssl.provider = null
	ssl.secure.random.implementation = null
	ssl.trustmanager.algorithm = PKIX
	ssl.truststore.location = null
	ssl.truststore.password = null
	ssl.truststore.type = JKS
	timeout.ms = 30000
	value.serializer = class org.apache.kafka.common.serialization.StringSerializer

2017-03-27 08:56:51.043  INFO   [main] --- o.a.k.clients.producer.KafkaProducer     : Closing the Kafka producer with timeoutMillis = 0 ms.
**2017-03-27 08:56:51.053 ERROR   [main] --- spacon.server                            : Uncaught exception on thread main**
**clojure.lang.ExceptionInfo: Error in component :kafka in system com.stuartsierra.component.SystemMap calling #'com.stuartsierra.component/start**
	at clojure.core$ex_info.invokeStatic(core.clj:4725)
	at clojure.core$ex_info.invoke(core.clj:4725)
	at com.stuartsierra.component$try_action.invokeStatic(component.cljc:119)
	at com.stuartsierra.component$try_action.invoke(component.cljc:116)
	at com.stuartsierra.component$update_system$fn__14882.invoke(component.cljc:139)
	at clojure.lang.ArraySeq.reduce(ArraySeq.java:114)
	at clojure.core$reduce.invokeStatic(core.clj:6703)
	at clojure.core$reduce.invoke(core.clj:6686)
	at com.stuartsierra.component$update_system.invokeStatic(component.cljc:135)
	at com.stuartsierra.component$update_system.doInvoke(component.cljc:129)
	at clojure.lang.RestFn.invoke(RestFn.java:445)
	at com.stuartsierra.component$start_system.invokeStatic(component.cljc:163)
	at com.stuartsierra.component$start_system.invoke(component.cljc:155)
	at com.stuartsierra.component$start_system.invokeStatic(component.cljc:161)
	at com.stuartsierra.component$start_system.invoke(component.cljc:155)
	at spacon.server$_main.invokeStatic(server.clj:99)
	at spacon.server$_main.doInvoke(server.clj:70)
	at clojure.lang.RestFn.invoke(RestFn.java:397)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at spacon.server.main(Unknown Source)
**Caused by: org.apache.kafka.common.KafkaException: Failed to construct kafka producer**
	at org.apache.kafka.clients.producer.KafkaProducer.<init>(KafkaProducer.java:338)
	at org.apache.kafka.clients.producer.KafkaProducer.<init>(KafkaProducer.java:178)
	at spacon.components.kafka.core$construct_producer.invokeStatic(core.clj:35)
	at spacon.components.kafka.core$construct_producer.invoke(core.clj:24)
	at spacon.components.kafka.core.KafkaComponent.start(core.clj:63)
	at com.stuartsierra.component$fn__14831$G__14825__14833.invoke(component.cljc:5)
	at com.stuartsierra.component$fn__14831$G__14824__14836.invoke(component.cljc:5)
	at clojure.lang.AFn.applyToHelper(AFn.java:154)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.lang.Var.applyTo(Var.java:700)
	at clojure.core$apply.invokeStatic(core.clj:659)
	at clojure.core$apply.invoke(core.clj:652)
	at com.stuartsierra.component$try_action.invokeStatic(component.cljc:117)
	... 18 common frames omitted
Caused by: java.lang.NullPointerException: null
	at org.apache.kafka.clients.ClientUtils.parseAndValidateAddresses(ClientUtils.java:40)
	at org.apache.kafka.clients.producer.KafkaProducer.<init>(KafkaProducer.java:300)
	... 30 common frames omitted

How to you solve this issue?

Thanks in advance,
Mat

--- a/README.md
+++ b/README.md
@@ -51,11 +51,12 @@ First startup the database container:
 docker-compose up -d postgis
 ```
 
-Then create the database and database user.  (You may need to wait a few seconds for the db container to fully start)
+Then create the database, database user, and schema.  (You may need to wait a few seconds for the db container to fully start)
 
 ```
 docker-compose run -e PGHOST=postgis -e PGUSER=postgres --rm postgis createuser spacon
 docker-compose run -e PGHOST=postgis -e PGUSER=postgres --rm postgis createdb spacon -O spacon
+docker-compose run -e PGHOST=postgis -e PGUSER=spacon --rm postgis psql -d spacon -c "CREATE SCHEMA spacon;"
 ```
 
 Then add the required extensions to our database.

